packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -243,16 +255,25 @@ export class SerializedStateManager {
 					);
 				}
 				assert(this.snapshot !== undefined, 0x8e5 /* no base data */);
-				const pendingRuntimeState = await runtime.getPendingLocalState(props);
+				const pendingRuntimeState = await runtime.getPendingLocalState({
+					...props,
+					sessionExpiryTimerStarted: this.updateSessionExpiryTime
+						? this.snapshotFetchedTime
+						: undefined,
+				});
 				const pendingState: IPendingContainerState = {
 					attached: true,
 					pendingRuntimeState,
 					baseSnapshot: this.snapshot.baseSnapshot,
 					snapshotBlobs: this.snapshot.snapshotBlobs,
 					savedOps: this.processedOps,
 					url: resolvedUrl.url,
-					// no need to save this if there is no pending runtime state
-					clientId: pendingRuntimeState !== undefined ? clientId : undefined,
+					// no need to save this if there is no pending runtime state ops
+					clientId:
+						(pendingRuntimeState as any).pending !== undefined ||
```

Comment:
we should not poke into the pendingRuntimeState, as that brings a lot of compatibility issues.